packages/cf.js/test/provider.spec.ts

```diff
@@ -173,4 +174,54 @@ describe("CF.js Provider", async () => {
     nodeProvider.simulateMessageFromNode(msg);
     nodeProvider.simulateMessageFromNode(msg);
   });
+
+  it("should load app instance details on-demand", async () => {
```

Comment:
This test, overall, seems really long. Can you break it up into several smaller tests?

Reply:
I don't think so :( Testing on-demand loading of details requires all of these steps. 

What I could do is make the test more readable by extracting methods